mirdata/phenicx_anechoic.py

```diff
@@ -0,0 +1,654 @@
+            self.targets[section] = Target(
+                sources=target_sections[section],
+                name=section,
+                instruments=instruments,
+                score_path=self.annotation_path,
+            )
+
+    def get_score(self, target):
+        """Get the score for a given target
+
+        Args:
+            target (str): name of the target e.g. violin
+
+        Returns:
+            (namedtuple, utils.EventData): the score in format 'start_times', 'end_times', 'event'
+
+        """
+        assert (
+            target in self.targets.keys()
+        ), 'target {} is not in the list of targets {}'.format(target, self.targets)
+        return self.targets[target].score
+
+    def get_original_score(self, target):
+        """Get the original score for a given target
+
+        Args:
+            target (str): name of the target e.g. violin
+
+        Returns:
+            (namedtuple, utils.EventData): the score in format 'start_times', 'end_times', 'event'
+
+        """
+        assert (
+            target in self.targets.keys()
+        ), 'target {} is not in the list of targets {}'.format(target, self.targets)
+        return self.targets[target].original_score
+
+    def get_audio_mix(self):
+        """Get the audio, sampling rate for the mix of sources
+
+        Returns:
+            (np.ndarray): the mono audio signal
+            (float): The sample rate of the audio file
+
+        """
+        return self.mix.audio, self.mix.rate
+
+    def get_audio_target(self, target):
+        """Get the audio, sampling rate for a given target
+
+        Args:
+            target (str): name of the target e.g. violin
+
+        Returns:
+            (np.ndarray): the audio signal
+            (float): The sample rate of the audio file
+
+        """
+        assert (
+            target in self.targets.keys()
+        ), 'target {} is not in the list of targets {}'.format(target, self.targets)
+        return self.targets[target].audio, self.targets[target].rate
+
+    def get_audio_source(self, source):
+        """Get the audio, sampling rate for a given source
+
+        Args:
+            source (str): name of the source e.g. violin1
+
+        Returns:
+            (np.ndarray): the audio signal
+            (float): The sample rate of the audio file
+
+        """
+        assert (
+            source in self.sources.keys()
+        ), 'source {} is not in the list of sources {}'.format(source, self.sources)
+        return self.sources[source].audio, self.sources[source].rate
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+        score_data = [
+            (self.targets[target].score, 'score-' + target) for target in self.targets
+        ]
+        original_score_data = [
+            (self.targets[target].original_score, 'score-' + target)
+            for target in self.targets
+        ]
+        metadata = {}
+        metadata['instruments'] = self.instruments
```

Comment:
I fixed this. btw we have some datasets which do not have audio but spectrograms or other descriptors (acousticbrainz, cofla).
how do we do it in this case? I think it's not ok to populate the duration field of an Event from the other field in the jams, like the audio. it can lead to weird errors. and exporting to jams should work even if a dataset does not have audio, to allow for the cases I mentioned.

Reply:
Yeah...I know and I agree. It's one of the things that jams has as a hard requirement in order for the schema to validate. For those datasets, I'd say for now put a dummy number as a bandaid while we think about what to do.